### PR TITLE
Provide better error message for invalid default export declaration

### PR DESCRIPTION
**Pull request by dczombera**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes ##6708 
| Patch: Bug Fix?          | n
| Major: Breaking Change?  | n
| Minor: New Feature?      | n
| Tests Added + Pass?      | Yes
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

I was not sure if an error message like the one suggested in the issue is consistent with the other error messages that babylon provides (is consistency even an issue when it comes to error messages?).
Apart from that, it's my first contribution to the project. I welcome any type of feedback. Thanks! :)

[REPL from build](https://babeljs.io/repl/build/7567/#?babili=false&browsers=&build=&builtIns=false&code_lz=KYDwDg9gTgLgBAE2AMwIYFcA28DGEB2AzvALICeAwhALaT7D7wC8cAFAJRxMB8cA3gCg4cKMBjoo-OAAMAPAgCWANzg5MqQoSYAiZBAjbuACQCSsgPSKl3aQG4BAX1tA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=true&presets=&prettier=true&targets=&version=7.0.0-beta.44%2Bpr.7717&envVersion=)

### CODE REVIEW

File: packages/babylon/src/parser/statement.js [hunk 1]
```diff
@@ -1408,6 +1408,16 @@ export default class StatementParser extends ExpressionParser {
...
       this.parseDecorators(false);
       return this.parseClass(expr, true, true);
+    } else if (
+      this.match(tt._let) ||
+      this.match(tt._const) ||
+      this.match(tt._var)
+    ) {
+      this.raise(
+        this.state.start,
+        "The `default` export can only take an expression;" +
```
Comment (nicolo-ribaudo) [inline packages/babylon/src/parser/statement.js]: It can also take function and class declarations.

Comment (dczombera) [reply packages/babylon/src/parser/statement.js]: So would you suggest to just add both to the error message?

Comment (existentialism) [reply packages/babylon/src/parser/statement.js]: Maybe just:

> Only expressions, functions or class declarations are allowed as the default export.

(Let's add tests for those too).

**Comment by dczombera** [comment]
Alright, I changed the error message as discussed and added tests for class declarations/expressions. :)

**Comment by hzoo** [comment]
Nice work, a great first PR!

**Comment by existentialism** [comment]
@dczombera thanks!

**Comment by nicolo-ribaudo** [comment]
An easy way to think about it is that named exports must have a name, while default exports not.

The real reason is that named exports export _bindings_, or variables (infact, if you modify an already exported variable the modules which import it will see the changes), while default exports export _values_.

**Comment by Kovensky** [comment]
But there's an asterisk... there's always an asterisk 😇 

If your default export is a `function` or `class` declaration (no parens wrapping them), then those will be mutable bindings, and reassigning the `function` or `class` name will update the default export.

It's also possible to make anything you want the default export by using `export { foo as default }`.

**Pull request merged**